stdlib/3/builtins.pyi

```diff
@@ -546,7 +546,8 @@ class tuple(Sequence[_T_co], Generic[_T_co]):
     def __rmul__(self, n: int) -> Tuple[_T_co, ...]: ...
     def count(self, x: Any) -> int: ...
     if sys.version_info >= (3, 5):
-        def index(self, x: Any, start: int = 0, end: int = 0) -> int: ...
+        def index(self, x: Any, start: int = None, end: int = None) -> int: ...  # TODO: __index__
+        # NOTE: tuple does not actually support None start/end, but it should to be compatible with Sequence.
```

Comment:
We should describe how tuple actually behaves, not how it's supposed to.

However, in this case I think you found at least a documentation bug in CPython: when you pass None it says `TypeError: slice indices must be integers or None or have an __index__ method`, and that error shouldn't mention None if None is not an acceptable value. Feel free to report that at bugs.python.org.

Unrelated problems with the existing code (outside the scope of this PR):
- The defaults should be `...`, not `0`.
- I feel like it would be more helpful to type the argument as `_T_co` instead of `Any`.